### PR TITLE
WIP - rework the ui to use typescript

### PR DESCRIPTION
**Pull request by richorama**

### CODE REVIEW

**Comment by SebastianStehle** [comment]
Are you also using a setup with webpack or so and a test server to have an easier time when developing?

**Comment by richorama** [comment]
Yes, it's webpack (CreateReactApp). I'll need to add a CORS rule in for local dev.

**Comment by SebastianStehle** [comment]
If it is not too much work I would also use hooks

**Comment by richorama** [comment]
yeah, this is quite a big rewrite, I'll keep it simple to start with and rework to use hooks as a next step.

File: ui/src/App.tsx [hunk 1]
```diff
@@ -1,15 +1,52 @@
...
+      hosts: []
+    }
+  }
+
   cancel?: () => void
 
   componentDidMount() {
     this.cancel = setIntervalDebounced(this.loadDataOnSchedule, 1000)
+
+    routie('', () => {
```
Comment (SebastianStehle) [inline ui/src/App.tsx:32]: I would use react-router like everybody else ;)

Comment (richorama) [reply ui/src/App.tsx:32]: Yes, I probably should! I despise using it though, almost as much as redux.

Comment (SebastianStehle) [reply ui/src/App.tsx:32]: Harsh words...why?

**Pull request closed without merging**